tests/laser/evm_testsuite/evm_test.py

```diff
@@ -64,40 +86,51 @@ def test_vmtest(
     # Act
     laser_evm.time = datetime.now()
 
-    # TODO: move this line below and check for VmExceptions when gas has been implemented
-    if post_condition == {}:
-        return
-
-    execute_message_call(
+    final_states = execute_message_call(
         laser_evm,
         callee_address=action["address"],
         caller_address=action["caller"],
         origin_address=action["origin"],
         code=action["code"][2:],
-        gas=action["gas"],
+        gas_limit=int(action["gas"], 16),
         data=binascii.a2b_hex(action["data"][2:]),
         gas_price=int(action["gasPrice"], 16),
         value=int(action["value"], 16),
+        track_gas=True,
     )
 
     # Assert
-
-    assert len(laser_evm.open_states) == 1
-
-    world_state = laser_evm.open_states[0]
-    model = get_model(next(iter(laser_evm.nodes.values())).states[0].mstate.constraints)
-
-    for address, details in post_condition.items():
-        account = world_state[address]
-
-        assert account.nonce == int(details["nonce"], 16)
-        assert account.code.bytecode == details["code"][2:]
-
-        for index, value in details["storage"].items():
-            expected = int(value, 16)
-            if type(account.storage[int(index, 16)]) != int:
-                actual = model.eval(account.storage[int(index, 16)])
-                actual = 1 if actual == True else 0 if actual == False else actual
-            else:
-                actual = account.storage[int(index, 16)]
-            assert actual == expected
+    if gas_used is not None and gas_used < 8000000:
+        # avoid gas usage larger than block gas limit
+        # this currently exceeds our estimations
+        gas_min_max = [
+            (s.mstate.min_gas_used, s.mstate.max_gas_used) for s in final_states
+        ]
+        gas_ranges = [g[0] <= gas_used <= g[1] for g in gas_min_max]
+        assert all(map(lambda g: g[0] <= g[1], gas_min_max))
+        assert any(gas_ranges)
+
+    if any((v in test_name for v in ["error", "oog"])):
+        # no more work to do if error happens or out of gas
```

Comment:
If this happens and also the post_condition={} only then the test is true.
if OOG/vmException is thrown and post_condtion!={} or post_condition={} and nothing is thrown then the test should raise an error.

Reply:
We don't get to see the internal exceptions, because they're caught [here](https://github.com/ConsenSys/mythril-classic/blob/5a7802643bf89d6c834da71dca25052d0f4ce16a/mythril/laser/ethereum/svm.py#L187). As mentioned earlier, we could cover that behaviour in unit tests for the `Instruction` class by testing closer to the `evaluate` method. Apart from some crazy mocking I wouldn't know how to dig it out of the scenario of the VM Tests.

Reply:
Should we be catching the vmException in mythril? all ethereum VMs directly raise that to say that something is wrong with the bytecode. @JoranHonig  what do you think?